### PR TITLE
Add travis integration

### PR DESCRIPTION
**Pull request by c-bata**
TODO

* [x] enable travis setting
* [x] Add badge
* [x] Pass CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: go
+
+go:
+  - 1.8
+  - 1.9
+  - 1.10
+  - 1.11
```
Comment (rhysd) [inline .travis.yml]: I think adding `tip` is useful to check next version Go does not break this library.

Comment (c-bata) [reply .travis.yml]: Sounds good!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+language: go
```
Comment (c-bata) [inline .travis.yml]: todo: add `sudo: false`

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        - diff <(goimports -d $(find . -type f -name '*.go' -not -path "./vendor/*")) <(echo -n "")
+        - make lint
+      go: "1.11.x"
+      os: osx
+
+script:
+  - make test
+  - make build
+
+allow_failures:
```
Comment (c-bata) [inline .travis.yml]: [todo] add windows in allow_failures because  Windows builds are in early access stage.

**Comment by c-bata** [comment]
I'm completely worn out 🎣

**Pull request merged**